### PR TITLE
Retrieve piece method

### PR DESCRIPTION
**Pull request by AUcod3r**
Added the retrieve_piece method to the game model.  Also modified the games controller update action and commented out some of the games.js code.  I tried to look at the ajax data being passed to the games_controller#update action and use that data as provided.  This code is NOT efficient... Within the method, I assign @pieces = Piece.all.  There has to be a better way, but I couldn't figure it out as of this commit with the current state of the overarching code.  I'm sure we'll need to refactor at some point but the method works and retrieves the correct piece.

### CODE REVIEW

**Comment by hoitomt** [comment]
@AUcod3r there are a lot of whitespace changes in this Pull Request. Please update your tab spaces in SublimeText (or whatever text editor you are using) to 2 spaces.

If you are using SublimeText you can navigate to the Default Settings (From the menu bar: Sublime Text -> Preferences -> Settings - Default). In that file search for tab_size and change it to 2. It looks like you have it set to 6 right now

**Comment by AUcod3r** [comment]
I'm using Atom and that is so weird.  I saw that on Github, went back and looked at my editor and it looks fine.  I will investigate...

**Comment by AUcod3r** [comment]
ok, @hoitomt , I corrected the whitespace in the model.

File: app/models/game.rb [hunk 1]
```diff
@@ -46,6 +47,19 @@ def create_pieces
     Piece.create(:type => 'King', :x_position => 4, :y_position => 7, user: black_player, game: self)
   end
 
+  def move_piece(origin, destination, id)
+    dest_x = destination[:x].to_i
+    dest_y = destination[:y].to_i
+    place = retrieve_piece(origin, id)
```
Comment (hoitomt) [inline app/models/game.rb]: typo - I think we want to call this `piece` rather than `place` :-)

File: app/models/game.rb [hunk 2]
```diff
@@ -46,6 +47,19 @@ def create_pieces
...
+  end
+
+  def retrieve_piece(origin, id)
+    orig_x = origin[:x].to_i
+    orig_y = origin[:y].to_i
+    @piece = Piece.all
+    return @piece.where(x_position: orig_x, y_position: orig_y, game_id: id)
```
Comment (hoitomt) [inline app/models/game.rb]: You can retrieve the piece without calling `Piece.all` first. Just call `Piece.where(x_position: orig_x, y_position: orig_y, game_id: id)`

**Comment by AUcod3r** [comment]
@hoitomt - I made the changes IAW your instruction.  You have become my father-figure...  :)

**Comment by hoitomt** [comment]
@AUcod3r LOL

I think this looks good now

**Comment by joehenriod** [comment]
Cool! I'll merge.

**Pull request merged**